### ISSUE TITLE
Dotnet 9.0 scaffolding no longer works - errors when doing NavigationManager.NavigateTo in Create.razor

### ISSUE THREAD

**Issue by coderdnewbie** [dotnet/aspnetcore#58951]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Perform the Dan Roth tutorial in https://www.youtube.com/watch?v=ZN-CcfEY3Z8

At 5:39 when the Create button is pressed, a failure happens in 

`NavigationManager.NavigateTo("/movies");`

This is the failure in VS2022 17.12:

Microsoft.AspNetCore.Components.NavigationException
  HResult=0x80131500
  Message=Exception_WasThrown
  Source=Microsoft.AspNetCore.Components.Server
  StackTrace:
   at Microsoft.AspNetCore.Components.Server.Circuits.RemoteNavigationManager.NavigateToCore(String uri, NavigationOptions options)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateToCore(String uri, Boolean forceLoad)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateTo(String uri, Boolean forceLoad, Boolean replace)
   at BlazorMovieCRUDDemo.Components.Pages.MoviePages.Create.<AddMovie>d__5.MoveNext() in C:\Users\ajayt\Desktop\Upgrade to dotnet 9 GA\BlazorMovieCRUDDemo\BlazorMovieCRUDDemo\Components\Pages\MoviePages\Create.razor:line 58
   at Microsoft.AspNetCore.Components.ComponentBase.<CallStateHasChangedOnAsyncCompletion>d__30.MoveNext()

This is from the autogenerated scaffolding performed exactly as Dan does it.

This used to work in dotnet 8.0, but now fails in dotnet 9.0.

Additionally, my dotnet 8.0 project when upgraded to dotnet 9.0 fails here as well.

### Expected Behavior

I expect this to work as it did in dotnet 8.0

### Steps To Reproduce

Perform the Dan Roth tutorial in https://www.youtube.com/watch?v=ZN-CcfEY3Z8

At 5:39 when the Create button is pressed, a failure happens in 

`NavigationManager.NavigateTo("/movies");`

This is scaffolded code in Create.razor in the MoviePages subfolder.

### Exceptions (if any)

This is the failure in VS2022 17.12:

Microsoft.AspNetCore.Components.NavigationException
  HResult=0x80131500
  Message=Exception_WasThrown
  Source=Microsoft.AspNetCore.Components.Server
  StackTrace:
   at Microsoft.AspNetCore.Components.Server.Circuits.RemoteNavigationManager.NavigateToCore(String uri, NavigationOptions options)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateToCore(String uri, Boolean forceLoad)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateTo(String uri, Boolean forceLoad, Boolean replace)
   at BlazorMovieCRUDDemo.Components.Pages.MoviePages.Create.<AddMovie>d__5.MoveNext() in C:\Users\ajayt\Desktop\Upgrade to dotnet 9 GA\BlazorMovieCRUDDemo\BlazorMovieCRUDDemo\Components\Pages\MoviePages\Create.razor:line 58
   at Microsoft.AspNetCore.Components.ComponentBase.<CallStateHasChangedOnAsyncCompletion>d__30.MoveNext()

### .NET Version

9.0.0

### Anything else?

![Image](https://github.com/user-attachments/assets/b17577da-1dfe-4fdf-88b6-278a5f083f6c)

**Comment by guardrex** [comment]
I just hit this when walking the tutorial for 9.0 updates ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/tutorials/movie-database-app/?view=aspnetcore-9.0

![Image](https://github.com/user-attachments/assets/50b58660-7ba4-4995-a5b1-ee748c0ce484)

Note @coderdnewbie that it isn't the "Even though this was selected as InteractiveServer in the dropdown" (**Interactive render mode**) setting that places the render mode on `HeadOutlet`/`Routes`. It's the **Interactivity location** setting. What you did by adding them manually made the interactivity location ***global*** instead of per page/component. The tutorial takes the per page/component approach because we have devs enable interactivity in the [last article of the tutorial series for the movies Index page](https://learn.microsoft.com/en-us/aspnet/core/blazor/tutorials/movie-database-app/part-8?view=aspnetcore-9.0&pivots=vs).

I agree with you ... it's a regression of some sort. This behavior wasn't happening in 8.0.

**Comment by guardrex** [comment]
@danroth27 ... How shall the tutorial deal with this issue in the meantime? The tutorial embraces per page/component until the last article of the series (Part 8 - *Add interactivity*), but this problem will stop devs dead 💀 in their tracks as soon as they hit scaffolding in Part 2 - *Add and scaffold a model*. If I refactor for global to unblock in Part 2, that's fine, but I'll also need to refactor per page/component/SSR language there and in other parts of the series, and I'll need to refactor Part 8 quite a bit. All of the workaround work will need to be undone later after this is resolved. I hope 🙏🤞🍀 that a quick patch can fix it in the framework to avoid all of that.

**Comment by coderdnewbie** [comment]
I definitely consider this a regression myself, as I have also found that if I try to upgrade from dotnet 8.0 to 9.0, the working solution from dotnet 8.0 fails in the same way, i.e. clicking the Create button causes the failure:

`NavigationManager.NavigateTo("/movies");`

https://github.com/danroth27 can you try it out and verify with an existing dotnet 8.0 solution and try to upgrade to dotnet 9.0. The same thing happens as I described in my initial post when clicking the Create button.

Thanks.

**Comment by javiercn** [comment]
This exception is thrown by the framework and handled by it. It's not a "new error" that's happening, it's simply that the debugger would not previously break on those exceptions.

**Comment by guardrex** [comment]
I didn't think that the debugger would change merely loading the 8.0 version of the tutorial sample app.

Anyway ... this definitely results in a poor tutorial experience for 9.0. I'll have to cover it, and I'll need to tell devs to uncheck the break-on checkbox to avoid it constantly happening as they work through the tutorial.

I'll wait to hear a confirmation from @danroth27 that that's what he wants me to do.

**Comment by danroth27** [comment]
@halter73 Where did you get in your investigations with the debugger team to prevent this exception from showing up?

**Comment by danroth27** [comment]
> This results in a poor tutorial experience for 9.0. I'll need to tell devs to uncheck the break-on checkbox to avoid it constantly happening as they work through the tutorial. I'll wait to hear a confirmation from @danroth27 that that's what he wants me to do.

@guardrex For now, I think we need to do this.

**Comment by coderdnewbie** [comment]
I agree that this is a poor experience, but unchecking the break-on box worked for me, so will close this issue.

@danroth27 I still think this needs to have the same behavior as dotnet 8.0, so that the demo tutorials still work for others. Should we be using 'Global' rather than 'Per Page/Per Component' in the checkbox?

**Comment by guardrex** [comment]
It will after the VS debugger is updated on a future release.

The written tutorial can't take that approach of going global because a latter step has the dev apply interactive SSR to a single component. It would be a large change to the tutorial series for a temporary gain. It's best to uncheck the box for now, and then I'll revert that guidance later when the next VS lands.

**Comment by guardrex** [comment]
@danroth27 ... WRT ...

> investigations with the debugger team to prevent this exception from showing up

Will the update land as a patch sooner, or will the update be for .NET 10?

With this issue closed, I have nothing to track for reverting the tutorial content.

**Comment by darthdaniel85** [comment]
> > This results in a poor tutorial experience for 9.0. I'll need to tell devs to uncheck the break-on checkbox to avoid it constantly happening as they work through the tutorial. I'll wait to hear a confirmation from [@danroth27](https://github.com/danroth27) that that's what he wants me to do.
> 
> [@guardrex](https://github.com/guardrex) For now, I think we need to do this.

Hi @danroth27, another problem is that even if you uncheck the break-on checkbox, you still get the exception if the navigation is performed within a try-catch... in those scenarios maybe this is an alternative: `catch (Exception e) when (e is not NavigationException)`.

**Comment by mongusheb** [comment]
It seems that blazer is not working properly in safari because of this issue.

**Comment by danroth27** [comment]
@mongusheb Since this issue is closed you should probably open a new issue with details about the issue you're seeing in Safari.

**Comment by danroth27** [comment]
Reopening as this is still happening with the latest .NET 9 and VS preview public builds.

**Comment by danroth27** [comment]
> It will after the VS debugger is updated on a future release.

@halter73 @gregg-miskelly Are we still waiting on a VS debugger update for this? If yes, what's the expected release? I tried out the latest public VS preview and I'm still seeing the debugger break on the NavigationException as reported here.

**Comment by gregg-miskelly** [comment]
There are no pending VS Debugger changes in this area. @halter73, is Daniel running with the latest batch of ASP.NET changes?

**Comment by danroth27** [comment]
@halter73 I tried this with .NET SDK 9.0.101 and also 9.0.200-preview.0.24575.35.

**Comment by danroth27** [comment]
Looks like this should be in the 9.0.1 patch release later this month: https://github.com/dotnet/aspnetcore/pull/58573
